blockchain/blockchain.go

```diff
@@ -854,4 +865,36 @@ func (bc *blockchain) replaceHeightAndHash(blk *Block) (uint64, hash.Hash32B, er
 	return tipHeight, tipHash, nil
 }
 
+func (bc *blockchain) SubscribeBlockCreation(ch chan *Block) error {
+	if bc.blocklistener == nil {
```

Comment:
In golang, you don't need to do this check, you should be able append to nil directly